src/python/pants/reporting/workunits.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+from typing import Tuple
+
+import requests
+from requests.exceptions import RequestException
+
+from pants.subsystem.subsystem import Subsystem
+
+
+class Workunits(Subsystem):
+  options_scope = 'workunits'
+
+  @classmethod
+  def register_options(cls, register):
+    register("--http-endpoint", type=str, help="Where to make HTTP requests to")
+
+  def make_http_request(self, workunits: Tuple[dict,...]):
+    options =  self.get_options()
+    http_endpoint = options.http_endpoint
+    data = { "workunits": workunits }
+
+    if not http_endpoint:
+      print("Streaming workunits are activated, but there is no specified HTTP endpoint to send them to")
+      return
+
+    try:
+      requests.post(http_endpoint, data=json.dumps(data))
+    except RequestException as e:
```

Comment:
By default this exception (or subclasses of it to be exact) are only raised on connection errors (i.e. read timeouts, SSL issues, etc..) not on HTTP error responses (HTTP 40x, HTTP 50x, etc..)
We should also log when those happen.
so you can do:
```python
 response = request.post(.....)
 if not response.ok:
    print(f"HTTP request failed {response.status_code} {response.text[:400]}")
```